lib/memo_wise.rb

```diff
@@ -423,6 +423,28 @@ def #{method_name}#{args_str}
           target.singleton_class.send(:private, method_name)
         end
       end
+
+      # Override [Module#instance_method](https://ruby-doc.org/core-3.0.0/Module.html#method-i-instance_method)
+      # to proxy the original `UnboundMethod#parameters` results. This proxying
+      # helps code that uses Ruby reflection over method parameters
+      def target.instance_method(symbol)
+        original_memo_wised_name = :"_memo_wise_original_#{symbol}"
+        curr_method = super
+
+        orig_method =
+          if symbol == :initialize
+            curr_method.super_method
+          elsif private_method_defined?(original_memo_wised_name)
+            super(original_memo_wised_name)
+          end
+
+        unless orig_method.nil?
```

Comment:
Could you add a comment explaining when `orig_method` will be `nil`? Is it just when there's no `initialize` method defined?

Reply:
I'll clarify the cases -- just returning `curr_method` when it was not a memo_wised method and not initialize -- can make it clearer

Reply:
Please re-review